newrelic/hooks/mlmodel_gemini.py

```diff
@@ -0,0 +1,560 @@
+# Copyright 2010 New Relic, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import sys
+import uuid
+
+import google
+
+from newrelic.api.function_trace import FunctionTrace
+from newrelic.api.time_trace import get_trace_linking_metadata
+from newrelic.api.transaction import current_transaction
+from newrelic.common.object_wrapper import wrap_function_wrapper
+from newrelic.common.package_version_utils import get_package_version
+from newrelic.core.config import global_settings
+
+GEMINI_VERSION = get_package_version("google-genai")
+EXCEPTION_HANDLING_FAILURE_LOG_MESSAGE = (
+    "Exception occurred in Gemini instrumentation: While reporting an exception "
+    "in Gemini, another exception occurred. Report this issue to New Relic "
+    "Support.\n "
+)
+RECORD_EVENTS_FAILURE_LOG_MESSAGE = (
+    "Exception occurred in Gemini instrumentation: Failed to record LLM events. "
+    "Please report this issue to New Relic Support.\n "
+)
+
+_logger = logging.getLogger(__name__)
+
+
+def wrap_embed_content_sync(wrapped, instance, args, kwargs):
+    transaction = current_transaction()
+    if not transaction:
+        return wrapped(*args, **kwargs)
+
+    settings = transaction.settings or global_settings()
+    if not settings.ai_monitoring.enabled:
+        return wrapped(*args, **kwargs)
+
+    # Framework metric also used for entity tagging in the UI
+    transaction.add_ml_model_info("Gemini", GEMINI_VERSION)
+    transaction._add_agent_attribute("llm", True)
+
+    # Obtain attributes to be stored on embedding events regardless of whether we hit an error
+    embedding_id = str(uuid.uuid4())
+
+    ft = FunctionTrace(name=wrapped.__name__, group="Llm/embedding/Gemini")
+    ft.__enter__()
+    linking_metadata = get_trace_linking_metadata()
+    try:
+        response = wrapped(*args, **kwargs)
+    except Exception as exc:
+        _record_embedding_error(transaction, embedding_id, linking_metadata, kwargs, ft, exc)
+        raise
+    ft.__exit__(None, None, None)
```

Comment:
Can we have a comment somewhere in here stating that the function trace is exited inside `_record_embedding_error` so it's less confusing? I'm sure this came from one of the other instrumentations, maybe check those and propagate the same comment?